caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditChannel.java

```diff
@@ -65,50 +65,41 @@ public HealthResult healthCheck()
             response = restClient.performRequest(healthRequest);
         } catch (final IOException ex) {
             logger.error("Error executing cluster health check request.", ex);
-            return new HealthResult(HealthStatus.UNHEALTHY, "Elasticsearch cluster is unhealthy");
+            return new HealthResult(HealthStatus.UNHEALTHY, "OpenSearch cluster is unhealthy");
         }
         return healthResponse(response.getEntity());
     }
 
     private HealthResult healthResponse(final HttpEntity httpEntity){
-        String healthResponse = "";
-        try{
+        String healthResponse = null;
+        try {
             healthResponse = EntityUtils.toString(httpEntity);
-        }catch (Exception e){
-            if(!validString(healthResponse)){
-                logger.error("HealthCheck response could not be processed", e);
+        } catch (final IOException e){
+            if(healthResponse.isEmpty() || healthResponse == null){
+                logger.error("HealthCheck response is null", e);
                 return new HealthResult(HealthStatus.UNHEALTHY, "HealthCheck response could not be processed");
             }
-            return new HealthResult(HealthStatus.UNHEALTHY, "Elasticsearch cluster is unhealthy");
+            return new HealthResult(HealthStatus.UNHEALTHY, "Cannot parse response from OpenSearch");
         }
```

Comment:
The if statement is still liable to throw a NPE.  Having said that I think the whole statement is redundant.  Just remove the if statement and log the exception.  The return at line 82 is proably enough.

Reply:
Leave the log statement in but update the message to be same as HealthResponse, same for below

Reply:
Just go with logger.error("Cannot parse response from OpenSearch", e);